haystack_experimental/components/image_converters/document_to_image.py

```diff
@@ -0,0 +1,178 @@
+    documents by extracting specific pages as images.
+
+    Documents are expected to have metadata containing:
+    - The `file_path_meta_field` is not present in the document metadata
+    - The file path does not point to an existing file when combined with `root_path`
+    - The file format is not among the supported image types
+    - For PDF files, the `page_number` is not present in the metadata
+    Otherwise, the component will raise a `ValueError` with a descriptive message.
+
+    Usage example:
+        ```python
+        converter = DocumentToImageContent(
+            file_path_meta_field="file_path",
+            root_path="/data/documents",
+            detail="high",
+            size=(800, 600)
+        )
+
+        documents = [
+            Document(content="Optional description of image.jpg", meta={"file_path": "image.jpg"}),
+            Document(content="Text content of page 1 of doc.pdf", meta={"file_path": "doc.pdf", "page_number": 1})
+        ]
+
+        result = converter.run(documents)
+        image_documents = result["image_documents"]
+        #
+        image_contents = result["image_contents"]
+        #
+        ```
+    """
+
+    def __init__(
+        self,
+        *,
+        file_path_meta_field: str = "file_path",
+        root_path: Optional[str] = None,
+        detail: Optional[Literal["auto", "high", "low"]] = None,
+        size: Optional[Tuple[int, int]] = None
+    ):
+        """
+        Initialize the DocumentToImageContent component.
+
+        :param file_path_meta_field: The metadata field in the Document that contains the file path to the image or PDF.
+        :param root_path: The root directory path where document files are located. If provided, file paths in
+            document metadata will be resolved relative to this path. If None, file paths are treated as absolute paths.
+        :param detail: Optional detail level of the image (only supported by OpenAI). Can be "auto", "high", or "low".
+            This will be passed to the created ImageContent objects.
+        :param size: If provided, resizes the image to fit within the specified dimensions (width, height) while
+            maintaining aspect ratio. This reduces file size, memory usage, and processing time, which is beneficial
+            when working with models that have resolution constraints or when transmitting images to remote services.
+        """
+        self.file_path_meta_field = file_path_meta_field
+        self.root_path = root_path or ""
+        self.detail = detail
+        self.size = size
+
+        # Initializing these converters will trigger the lazy import checks.
+        self._file_to_image_converter = ImageFileToImageContent(detail=detail, size=size)
+        self._pdf_to_image_converter = PDFToImageContent(detail=detail, size=size)
+
+    @component.output_types(image_documents=List[Document], image_contents=List[ImageContent])
```

Comment:
I have the impression that now we are returning the original list of Documents, unless we encounter errors during the process. In this case, we might simplify this component to only return a list of `ImageContent`.
Makes sense or am I missing some details?